File: Bucket Sort/BucketSort.cpp
#include <iostream>
using namespace std;

void bucketSort(float arr[], int n) {
    for(int i = 0; i < n - 1; i++) {
        for(int j = 0; j < n - i - 1; j++) {
            if(arr[j] > arr[j + 1]) {
                float temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main() {
    float arr[] = {0.42, 0.32, 0.23, 0.52, 0.25, 0.47};
    int n = 6;

    bucketSort(arr, n);

    for(int i = 0; i < n; i++)
        cout << arr[i] << " ";
}

File: Bucket Sort/BucketSort.java
class BucketSort {

    static void bucketSort(float arr[]) {
        int n = arr.length;

        for(int i = 0; i < n - 1; i++) {
            for(int j = 0; j < n - i - 1; j++) {
                if(arr[j] > arr[j + 1]) {
                    float temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        float arr[] = {0.42f, 0.32f, 0.23f, 0.52f, 0.25f, 0.47f};

        bucketSort(arr);

        for(float num : arr)
            System.out.print(num + " ");
    }
}
